tokio-console/src/config.rs

```diff
@@ -238,3 +312,86 @@ impl FromStr for RetainFor {
         }
     }
 }
+
+// === impl ColorToggles ===
+
+impl ColorToggles {
+    pub fn color_durations(&self) -> bool {
+        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
+    }
+
+    pub fn color_terminated(&self) -> bool {
+        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
+    }
+}
+
+// === impl ColorToggles ===
+
+impl ConfigFile {
+    fn from_config(path: ConfigPath) -> color_eyre::Result<Option<Self>> {
+        let config = path
+            .into_path()
+            .and_then(|path| fs::read_to_string(path).ok())
+            .map(|raw| toml::from_str::<ConfigFile>(&raw))
+            .transpose()
+            .wrap_err_with(|| format!("failed to parse {:?}", path.into_path()))?;
```

Comment:
A small note: I think the path will be formatted nicer if we use [`Path::display`](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.display) here instead of its `Debug` implementation:
```suggestion
            .wrap_err_with(|| format!("failed to parse {}", path.into_path().display()))?;
```